packages/instant-apps-components/src/components/instant-apps-social-share/instant-apps-social-share.test.ts

```diff
@@ -1,319 +1,253 @@
+
+  const testCases: allTests = [
+    [
+      'popover',
+      function createPopover() {
+        let ss = document.createElement('instant-apps-social-share');
+        return ss;
+      },
+    ],
+    [
+      'inline',
+      function createInline() {
+        let ss = document.createElement('instant-apps-social-share');
+        ss.setAttribute('mode', 'inline');
+        return ss;
+      },
+    ],
+  ];
+
+  testCases.forEach(arr => {
+    let element;
+    let shadow;
+    const [elem, getElem] = arr;
+    const socialShare = getElem();
+
+    const isInline = elem === 'inline';
+    describe(`${elem}`, async () => {
+      beforeEach(async () => {
+        document.body.append(socialShare);
+        await new Promise(resolve => requestIdleCallback(resolve));
+        element = document.querySelector('instant-apps-social-share');
+        shadow = await waitForShadowRoot(element);
+      });
+      test('default', () => {
+        expect(element).toBeTruthy();
+        expect(element.getAttribute('mode')).toBe(`${elem}`);
+      });
+      test('embed prop', async () => {
+        expect(element).toBeTruthy();
+        expect(element?.embed).toBe(false);
+        socialShare.embed = true;
+        document.body.append(socialShare);
+        await new Promise(resolve => requestIdleCallback(resolve));
+
+        expect(element?.embed).toBe(true);
+        expect(element.inlineCopyEmbedOpened).toBe(false);
+      });
+      describe.runIf(!isInline)('popover render', async () => {
+        test('check renderButton', async () => {
+          const button = shadow.querySelector("calcite-button[id='shareButton']");
+          expect(button).toBeTruthy();
+        });
+        test('check popover', async () => {
+          const popover = shadow.querySelector(`calcite-popover[label='${SocialShare_T9n.share.label}']`);
+          expect(popover).toBeTruthy();
+        });
+        test('shareButtonColor', async () => {
+          expect(element).toBeTruthy();
+          expect(element?.shareButtonColor).toBe('neutral');
+          socialShare.shareButtonColor = 'inverse';
+          document.body.append(socialShare);
+          await new Promise(resolve => requestIdleCallback(resolve));
+          expect(element?.shareButtonColor).toBe('inverse');
+          const kind = shadow.querySelector("calcite-button[id='shareButton']").getAttribute('kind');
+          expect(kind).toBe(socialShare.shareButtonColor);
+        });
+
+        test('shareButtonType', async () => {
+          expect(element).toBeTruthy();
+          expect(element?.shareButtonType).toBe('button');
+          const button = shadow.querySelector("calcite-button[id='shareButton']");
+          expect(button).toBeTruthy();
+
+          socialShare.shareButtonType = 'action';
+          document.body.append(socialShare);
+          await new Promise(resolve => requestIdleCallback(resolve));
+          const action = shadow.querySelector("calcite-action[id='shareButton']");
```

Comment:
why not use getElementById?

Reply:
I don't think you can directly use getElementById to access elements within a shadow root